Vulnerabilities/vulnerable_components.md

```diff
@@ -0,0 +1,19 @@
+# [Using Components with Known Vulnerabilities](https://owasp.org/www-project-top-ten/2017/A9_2017-Using_Components_with_Known_Vulnerabilities)
+
+It is easy to take for granted that the libraries one uses and relies on are secure, reliable, and well maintained. Even libraries whose authors have the best intentions, however, can have vulnerabilities slip in. Responsible library maintainers will normally quickly put out a patch, but if the end user relying on the package doesn't update their version to the new on including the patch, the will be stuck with the vulnerability. 
+
+## Our Implementation
+
+There are many places where vulnerable components can exist, the OS, the web/application server, the DBMS, libraries, etc. We found a vulnerability in a recent version of Bootstrap, a library we were already using, and configured our site so that the insecure version called an earlier, vulnerable version of Bootstrap (4.0.0) while our secure site used the most up-to-date available version of Bootstrap (4.3.1). The vulnerability in question is [CVE-2019-8331](https://github.com/advisories/GHSA-9v3m-8fp8-mj99). We then implemented a simple mechanism to trigger the vulnerability in the earlier version to demonstrate the weakness and also to verify the fix.
```

Comment:
> can exist,

can exist:

Reply:
> We then implemented a simple mechanism to trigger the vulnerability in the earlier version to demonstrate the weakness and also to verify the fix.

I think we should more specifically explain what happened in our exploit:
The vulnerable bootstrap version 4.0.0 will parse and execute scripts that are injected into the data-template attribute of the tooltip and popover plugins which enables cross-site scripting. The 4.3.1 version has patched this.

Reply:
Yeah, that's a good idea. I was in a rush to write this first draft, I'll revisit it when I've got more time to spend on it this weekend.

Reply:
Ok I rewrote it, would appreciate another once-over. Thanks!